### PR TITLE
Bump SDK in old release of build_modules

### PR DESCRIPTION
**Pull request by natebosch**
Closes https://github.com/dart-lang/sdk/issues/51128

SDK version `2.19.0` introduces a static error that was missing from earlier SDK versions. Upgrading to language version `2.18` or greater brings a language change with improved inference which resolves the issue. Upgrade to `2.19.0` so that the backported change impacts the fewest users possible.

### CODE REVIEW

**Comment by natebosch** [comment]
My plan is to wait for approval, publish from this commit, resolve conflicts with a merge _from_ master, then merge this PR to master with a merge commit. I'll have to temporarily enable merge commits - any concerns with that @kevmoo ?

**Comment by kevmoo** [comment]
@natebosch – SGTM!

**Review by jakemac53 — APPROVED** [review]

**Comment by natebosch** [comment]
@jodinathan do I understand correctly that you found another solution to this?

Would it still be useful to have a 2.19.0 compatible version of  `build_modules` version `4.0.6` published, or can we close this?

**Comment by jodinathan** [comment]
yes you can @natebosch.

We just migrated angular to analyzer 5 using ngdart (thanks to the community heroic effort) as base and the test were successful. So this issue isn't blocking us from upgrading to 2.19 anymore.

It has been years and you guys are the same incredible as before. Thanks for the help!

**Pull request closed without merging**
